specs/AppNotifications/BadgeNotificationSpec/BadgeNotifications-spec.md

```diff
@@ -0,0 +1,130 @@
+# Badge notifications in Windows App SDK
+
+- [1. Background](#background)
+- [2. Limitations of the Windows SDK APIs](#limitations-of-the-windows-sdk-apis)
+- [3. API proposal](#api-proposal)
+- [4. Remarks](#remarks)
+- [5. IDL details](#idl-details)
+
+## Background
+
+Badges are a type of notifications indicating a summary or status information specific to an app. It is represented either by a number or a glyph sitting over the top-right corner of the taskbar app icon.
+
+Here is an example of a numeric badge posted by Microsoft To-Do List:
+
+![Example of a numeric badge in Microsoft To-Do List](NumericBadge.png)
+
+Another example of a badge, in this case featuring a new message glyph posted by Outlook:
+
+![Example of a busy status glyph badge in Microsoft Teams](GlyphBadge.png)
+
+## Limitations of the Windows SDK APIs
+
+Badge notification Windows SDK APIs have three main limitations:
+
+- They don't support Win32 unpackaged apps.
+
+- The Badge Windows SDK APIs expect an XML payload that contains the information about the badge to pin, even though the XML is a single line of content. There are no APIs where developers can directly send the number or specify the glyph type.
+
+- As of today, Badge APIs support a limited set of glyphs. If the app doesn't find any suitable glyph to convey their status, there is no option to pin a custom overlay.
+
+More information about Windows SDK Badge API usage can be found [here](https://learn.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/badges#create-a-badge).
+
+## API proposal
+
+As part of bringing badges into Windows App SDK, we propose addressing the issues explained the previous section. With this proposal, developers will be able to pin and clear badges writing less code than what is required in the Windows SDK.
+
+It's worth mentioning this proposal covers local badge notifications. Cloud badges are out of the proposal scope.
+
+### 1. Posting a numeric badge
+
+```cpp
+BadgeNotificationManager::ShowNumber(4);
```

Comment:
@hulumane we can't have multiple ctors with the same arity. I still haven't discarded the idea of having a Badge object because of the HWND and description required for the Taskbar API.